resources/fetchersManager/registry.go

```diff
@@ -42,21 +44,39 @@ type registeredFetcher struct {
 	c []fetching.Condition
 }
 
-func NewFetcherRegistry(log *logp.Logger) FetchersRegistry {
-	return &fetchersRegistry{
+func NewFetcherRegistry(log *logp.Logger, cfg config.Config) FetchersRegistry {
+	reg := &fetchersRegistry{
 		log: log,
 		reg: make(map[string]registeredFetcher),
 	}
+	return reg
+}
+
+// RegisterFetchers registers entire list of parsed fetchers
+func (r *fetchersRegistry) RegisterFetchers(fetchers []*ParsedFetcher) error {
+	for _, p := range fetchers {
+		c, err := Factories.getConditions(r.log, p.name)
+		if err != nil {
+			return fmt.Errorf("RegisterFetchers error in getConditions for factory %s skipping Register due to: %v", p.name, err)
+		}
+
+		err = r.Register(p.name, p.f, c...)
+		if err != nil {
+			return fmt.Errorf("RegisterFetchers error in Register for factory %s skipping Register due to: %v", p.name, err)
+		}
+	}
+
+	return nil
 }
 
 // Register registers a Fetcher implementation.
-func (r *fetchersRegistry) Register(key string, f fetching.Fetcher, c ...fetching.Condition) error {
-	r.log.Infof("Registering new fetcher: %s", key)
-	if _, ok := r.reg[key]; ok {
-		return fmt.Errorf("fetcher key collision: %q is already registered", key)
+func (r *fetchersRegistry) Register(fetcherName string, f fetching.Fetcher, c ...fetching.Condition) error {
```

Comment:
Do `Register` now need to stay public method?
I guess it can go private, isn't it?